paf-mvp-core-js/src/express/key-store.ts

```diff
@@ -1,26 +1,31 @@
 import { fromIdentityResponse, KeyInfo } from '@core/crypto/identity';
 import { GetIdentityRequestBuilder } from '@core/model/identity-request-builder';
-import { GetIdentityResponse } from '@core/model/generated-model';
+import { GetIdentityResponse, Timestamp } from '@core/model/generated-model';
 import { PublicKey, publicKeyFromString } from '@core/crypto/keys';
-import axios, { Axios, AxiosRequestConfig, AxiosResponse } from 'axios';
+import axios, { AxiosRequestConfig, AxiosResponse } from 'axios';
+import { getTimeStampInSec } from '@core/timestamp';
 
-type PublicKeyInfo = KeyInfo & { publicKeyObj: PublicKey };
+export type PublicKeyInfo = KeyInfo & { publicKeyObj: PublicKey };
 
 export class PublicKeyStore {
   protected cache: { [domain: string]: PublicKeyInfo } = {};
-  protected s2sClient: Axios;
 
-  constructor(s2sOptions?: AxiosRequestConfig) {
-    this.s2sClient = axios.create(s2sOptions);
-  }
+  constructor(
+    s2sOptions?: AxiosRequestConfig,
+    protected s2sClient = axios.create(s2sOptions),
+    protected timestampProvider: () => Timestamp = getTimeStampInSec
+  ) {}
 
   async getPublicKey(domain: string): Promise<PublicKeyInfo> {
-    const nowTimestampSeconds = new Date().getTime() / 1000;
+    const nowTimestampSeconds = this.timestampProvider();
 
     const existingKey = this.cache[domain];
 
     // Make sure this key is not out dated. If so, then consider no cache value and request it from identity endpoint
-    if (existingKey && nowTimestampSeconds < existingKey.end.getTime()) {
+    if (
+      existingKey &&
+      (existingKey.endTimestampInSec === undefined || nowTimestampSeconds < existingKey.endTimestampInSec)
```

Comment:
Bugfix!
1. was triggering error if the public key doesn't have an end date
2. was comparing milliseconds with seconds! 😱